### PR TITLE
Added the use of cloud-init or post-install script to  create a new bare metal device

### PR DESCRIPTION
**Pull request by rafikrafael**
Added the use of script property to deploy a bare metal device with Cloud-init and Post-Install script

### CODE REVIEW

File: examples/bare_metal_device/main.tf [hunk 1]
```diff
@@ -13,6 +13,7 @@ resource "hivelocity_bare_metal_device" "demo" {
   location_name = "DAL1"
   hostname = "hivelocity.terraform.test"
   tags = ["hello", "world"]
+  script = "#cloud-config\npackage_update: true\npackages:\n - vim"
```
Comment (davinirjr) [inline examples/bare_metal_device/main.tf]: I think we should go for a `file` example. 
Please check:
https://www.terraform.io/docs/language/functions/file.html

File: examples/readme_example/main.tf [hunk 2]
```diff
@@ -34,4 +34,5 @@ resource "hivelocity_bare_metal_device" "tampa_server" {
     location_name = "${data.hivelocity_product.tampa_product.location}"
     hostname = "hivelocity.terraform.test"
     tags = ["hello", "world"]
+    script = "#cloud-config\npackage_update: true\npackages:\n - vim"
```
Comment (davinirjr) [inline examples/readme_example/main.tf]: Same here.

**Review by davinirjr — COMMENTED** [review]
Nice work, buddy. I left some comments regarding the use of the function `file`.

File: examples/bare_metal_device/main.tf (hunk 1 above)
Comment (rafikrafael) [reply examples/bare_metal_device/main.tf]: my fault , I will improve this file example load, thanks Davinir

**Review by mvellasco — APPROVED** [review]
LGTM

File: README.md [hunk 3]
```diff
@@ -87,15 +87,20 @@ data "hivelocity_product" "tampa_product" {
 resource "hivelocity_bare_metal_device" "tampa_server" {
     product_id = "${data.hivelocity_product.tampa_product.product_id}"
     os_name = "CentOS 7.x"
-    location_name = "${data.hivelocity_product.tampa_product.location}"
+    location_name = "${data.hivelocity_product.tampa_product.data_center}"
```
Comment (davinirjr) [inline README.md:90]: Do this change is required? I am asking because this is a breaking change, right?

Comment (rafikrafael) [reply README.md:90]: yes, the name location was changed to data_center on the endpoint. Was changed the product/list to use the FastInventoryService.flat_inventory and this returns "data_center".

**Review by davinirjr — APPROVED** [review]
There is a comment out there regarding breaking changes. LGTM.

**Pull request merged**